operate/services/manage.py

```diff
@@ -2061,11 +2091,117 @@ def refill_requirements(  # pylint: disable=too-many-locals
+        self, service_config_id: str, chain: str
+    ) -> int:
+        """Computes the bonded olas: current agent bonds and current security deposit"""
+
+        service = self.load(service_config_id=service_config_id)
+        chain_config = service.chain_configs[chain]
+        ledger_config = chain_config.ledger_config
+        user_params = chain_config.chain_data.user_params
+        wallet = self.wallet_manager.load(ledger_config.chain.ledger_type)
+        master_safe = wallet.safes[Chain(chain)]
+
+        ledger_api = wallet.ledger_api(chain=ledger_config.chain, rpc=ledger_config.rpc)
+
+        service_id = chain_config.chain_data.token
+        if service_id == NON_EXISTENT_TOKEN:
+            return 0
+
+        os.environ["CUSTOM_CHAIN_RPC"] = ledger_config.rpc
+        sftxb = self.get_eth_safe_tx_builder(ledger_config=ledger_config)
+        staking_params = sftxb.get_staking_params(
+            staking_contract=STAKING[ledger_config.chain][
+                user_params.staking_program_id
+            ],
+        )
+        service_registry_address = staking_params["service_registry"]
+        service_registry_token_utility_address = staking_params[
+            "service_registry_token_utility"
+        ]
+        service_registry = registry_contracts.service_registry.get_instance(
+            ledger_api=ledger_api,
+            contract_address=service_registry_address,
+        )
+        service_registry_token_utility = (
+            registry_contracts.service_registry_token_utility.get_instance(
+                ledger_api=ledger_api,
+                contract_address=service_registry_token_utility_address,
+            )
+        )
+
+        service_owner = service_registry.functions.ownerOf(service_id).call()
+
+        # If master safe is service owner, the service is not staked
+        if service_owner.lower() == master_safe.lower():
+            return 0
+
+        service_info = service_registry.functions.getService(service_id).call()
+        agent_ids = service_info[7]
+
+        agent_bonds = 0
+        for agent_id in agent_ids:
+            num_agent_instances = service_registry.functions.getInstancesForAgentId(
+                service_id, agent_id
+            ).call()[0]
+            agent_bond = service_registry_token_utility.functions.getAgentBond(
+                service_id, agent_id
+            ).call()
+            agent_bonds += num_agent_instances * agent_bond
+
+        security_deposit = 0
+        service_state = service_info[6]
+        if (
+            OnChainState.ACTIVE_REGISTRATION
+            <= service_state
+            < OnChainState.TERMINATED_BONDED
+        ):
+            security_deposit = (
+                service_registry_token_utility.functions.mapServiceIdTokenDeposit(
+                    service_id
+                ).call()[1]
+            )
+
+        return agent_bonds + security_deposit
+
+    def _compute_olas_requirements(  # pylint: disable=too-many-locals
```

Comment:
@kupermind Please review this method. The output should be the olas requirements to correctly set up the service.